sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -977,6 +977,33 @@ public GroupByQuery toGroupByQuery()
       // Cannot handle zero or negative limits.
       return null;
     }
+    Map<String, Object> theContext = plannerContext.getQueryContext();
+
+    Granularity queryGranularity = null;
+
+    if (!grouping.getDimensions().isEmpty()) {
+      for (DimensionExpression dimensionExpression : grouping.getDimensions()) {
```

Comment:
Should we exclude the `dimensionExpression` from the dimensionSpecs that is passed to the constructor of `GroupByQuery` if it's a time_floor function? It seems duplicate since we can now pass the right granularity.

Reply:
See the first reply.